examples_tests/20.Megatexture/main.cpp

```diff
@@ -1,44 +1,378 @@
 #define _IRR_STATIC_LIB_
+#include <iostream>
+#include <cstdio>
 #include <irrlicht.h>
 
-#include "../ext/ScreenShot/ScreenShot.h"
+//! I advise to check out this file, its a basic input handler
 #include "../common/QToQuitEventReceiver.h"
+#include <irr/asset/ITexturePacker.h>
+#include <irr/asset/CMTLPipelineMetadata.h>
 
+//#include "../../ext/ScreenShot/ScreenShot.h"
 using namespace irr;
 using namespace core;
 
+#include "irr/irrpack.h"
+struct STextureData
+{
+    uint16_t pgTab_x;
+    uint16_t pgTab_y;
+    uint16_t width;
+    uint16_t height;
+} PACK_STRUCT;
+#include "irr/irrunpack.h"
+
+STextureData getTextureData(const asset::ICPUImage* _img, asset::ICPUTexturePacker* _packer)
+{
+    STextureData texData;
+    core::vector2df_SIMD szUnorm16(_img->getCreationParameters().extent.width, _img->getCreationParameters().extent.height);
+    szUnorm16 /= core::vector2df_SIMD(_packer->getPageTable()->getCreationParameters().extent.width);
+    szUnorm16 *= core::vector2df_SIMD(0xffffu);
+
+    texData.width = szUnorm16.x;
+    texData.height = szUnorm16.y;
+
+    asset::IImage::SSubresourceRange subres;
+    subres.baseMipLevel = 0u;
+    subres.levelCount = core::findLSB(core::roundDownToPoT<uint32_t>(std::max(texData.width, texData.height))) + 1;
+
+    auto pgTabCoords = _packer->pack(_img, subres);
+    assert((pgTabCoords!=asset::ITexturePacker::page_tab_offset_invalid()).all());
+
+    core::vector2df_SIMD pgTabUnorm16(pgTabCoords.x, pgTabCoords.y);
+    pgTabUnorm16 /= core::vector2df_SIMD(_packer->getPageTable()->getCreationParameters().extent.width, _packer->getPageTable()->getCreationParameters().extent.height, 1.f, 1.f);
+    pgTabUnorm16 *= core::vector2df_SIMD(0xffffu);
+    texData.pgTab_x = pgTabUnorm16.x;
+    texData.pgTab_y = pgTabUnorm16.y;
+
+    return texData;
+}
 
-class SimpleCallBack : public video::IShaderConstantSetCallBack
+constexpr const char* GLSL_VT_TEXTURES = //also turns off set3 bindings (textures) because they're not needed anymore as we're using VT
+R"(
+#ifndef _NO_UV
+layout (set = 0, binding = 0) usampler2D pgTabTex;
+layout (set = 0, binding = 1) sampler2DArray physPgTex;
+#endif
+#define _IRR_FRAG_SET3_BINDINGS_DEFINED_
```

Comment:
but you're using `set=0` above?

Reply:
yea see the comment in top line. It disables set3 as an extra because they're not needed

Reply:
ah ok